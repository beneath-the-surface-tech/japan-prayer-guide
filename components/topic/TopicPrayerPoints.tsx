import React from "react";
import { Container, Card } from "react-bootstrap";
import {FaPrayingHands} from 'react-icons/fa'
import { IconContext } from "react-icons/lib";
import { BsDownload } from "react-icons/bs";
import Button from "react-bootstrap/Button";

interface Props {
    prayerPoints: string[]
}

export default function TopicPrayerPoints({prayerPoints}:Props) {
    return (
        <Container className="d-flex justify-content-center px-6" id="topic-prayer">
            <Card className="my-4 shadow d-lg-flex flex-lg-row">
                {/* temporary height for video, before we finalize videos */}
                <Card.Img variant="top" src="/bamboo.jpeg" style={{height:'300px', maxWidth:'600px'}}></Card.Img>
                <Card.Body style={{maxWidth:'600px'}}>
                    <Card.Text className="px-2 pb-3 fs-2 fst-italic fw-bold border-bottom border-grey d-flex justify-content-between align-items-center">
                        Prayer Points
                        <IconContext.Provider value={{size:'30px'}}>
                            <BsDownload className="text-secondary fw-bold" style={{cursor: 'pointer'}}>
                            </BsDownload>
                        </IconContext.Provider>
                    </Card.Text>
                    <Card.Text className="d-flex align-items-center gap-2 text-secondary">
                        <IconContext.Provider value={{size:'20px'}}>
                            <FaPrayingHands></FaPrayingHands>
                        </IconContext.Provider>
                        Pray For
                    </Card.Text>
                    <ul className="fs-5">
                        {prayerPoints.map((point:string) => 
                            <li key={point} className="my-3">{point}</li>
                        )}
                    </ul>
                </Card.Body>
            </Card>
        </Container>
    )
}