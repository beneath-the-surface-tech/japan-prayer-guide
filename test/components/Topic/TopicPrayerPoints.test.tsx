import { render, screen } from '@testing-library/react';
import TopicPrayerPoints from '../../../components/topic/TopicPrayerPoints';

describe("Topic Prayer Points", () => {
    test("Renders a section with the right text inside", () => {
        const testTextArray = ["Hey how's it going", "I'm doing fine", "Thanks"]
        const testTitle = "Topic Prayer Points"
        const testSubtitle = "Topic Prayer Points Sub"
        render(<TopicPrayerPoints prayerPoints={testTextArray} title={testTitle} subtitle={testSubtitle}/>)
        const TopicPrayerPointsCont = screen.getByTestId("topic-points-container")
        const TopicPrayerPointsTitle = screen.getByTestId("topic-points-title")
        const TopicPrayerPointsPoints = screen.getByTestId("topic-points-points").children

        expect(TopicPrayerPointsCont).toHaveClass("d-flex", "container", "px-6")

        expect(TopicPrayerPointsTitle).toHaveTextContent(testTitle)
        expect(TopicPrayerPointsTitle).toHaveClass("px-2", "pb-3", "d-flex")

        expect(TopicPrayerPointsPoints.length).toBe(3)
        expect(TopicPrayerPointsPoints[0]).toHaveTextContent(testTextArray[0])
        expect(TopicPrayerPointsPoints[0]).toHaveClass("my-3")
        expect(TopicPrayerPointsPoints[1]).toHaveTextContent(testTextArray[1])
        expect(TopicPrayerPointsPoints[1]).toHaveClass("my-3")
        expect(TopicPrayerPointsPoints[2]).toHaveTextContent(testTextArray[2])
        expect(TopicPrayerPointsPoints[2]).toHaveClass("my-3")
    })
})