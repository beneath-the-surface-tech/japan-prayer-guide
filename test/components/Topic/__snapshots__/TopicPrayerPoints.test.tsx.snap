// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic Prayer Points Snapshot test 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-center px-6 container"
        data-testid="topic-points-container"
        id="topic-prayer"
      >
        <div
          class="my-4 shadow d-md-flex flex-md-row card"
        >
          <img
            class="card-img-top"
            src="/bamboo.jpeg"
            style="height: 300px; max-width: 600px;"
          />
          <div
            class="card-body"
            style="max-width: 600px;"
          >
            <p
              class="px-2 pb-3 fs-2 fst-italic fw-bold border-bottom border-grey d-flex justify-content-between align-items-center card-text"
              data-testid="topic-points-title"
            >
              Prayer Points
              <svg
                class="text-secondary fw-bold"
                fill="currentColor"
                height="30px"
                stroke="currentColor"
                stroke-width="0"
                style="cursor: pointer;"
                viewBox="0 0 16 16"
                width="30px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"
                />
                <path
                  d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"
                />
              </svg>
            </p>
            <p
              class="d-flex align-items-center gap-2 text-secondary card-text"
            >
              <svg
                fill="currentColor"
                height="20px"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="20px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M272 191.91c-17.6 0-32 14.4-32 32v80c0 8.84-7.16 16-16 16s-16-7.16-16-16v-76.55c0-17.39 4.72-34.47 13.69-49.39l77.75-129.59c9.09-15.16 4.19-34.81-10.97-43.91-14.45-8.67-32.72-4.3-42.3 9.21-.2.23-.62.21-.79.48l-117.26 175.9C117.56 205.9 112 224.31 112 243.29v80.23l-90.12 30.04A31.974 31.974 0 0 0 0 383.91v96c0 10.82 8.52 32 32 32 2.69 0 5.41-.34 8.06-1.03l179.19-46.62C269.16 449.99 304 403.8 304 351.91v-128c0-17.6-14.4-32-32-32zm346.12 161.73L528 323.6v-80.23c0-18.98-5.56-37.39-16.12-53.23L394.62 14.25c-.18-.27-.59-.24-.79-.48-9.58-13.51-27.85-17.88-42.3-9.21-15.16 9.09-20.06 28.75-10.97 43.91l77.75 129.59c8.97 14.92 13.69 32 13.69 49.39V304c0 8.84-7.16 16-16 16s-16-7.16-16-16v-80c0-17.6-14.4-32-32-32s-32 14.4-32 32v128c0 51.89 34.84 98.08 84.75 112.34l179.19 46.62c2.66.69 5.38 1.03 8.06 1.03 23.48 0 32-21.18 32-32v-96c0-13.77-8.81-25.99-21.88-30.35z"
                />
              </svg>
              Sub
            </p>
            <ul
              class="fs-5"
              data-testid="topic-points-points"
            >
              <li
                class="my-3"
              >
                ab
              </li>
              <li
                class="my-3"
              >
                cd
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-center px-6 container"
      data-testid="topic-points-container"
      id="topic-prayer"
    >
      <div
        class="my-4 shadow d-md-flex flex-md-row card"
      >
        <img
          class="card-img-top"
          src="/bamboo.jpeg"
          style="height: 300px; max-width: 600px;"
        />
        <div
          class="card-body"
          style="max-width: 600px;"
        >
          <p
            class="px-2 pb-3 fs-2 fst-italic fw-bold border-bottom border-grey d-flex justify-content-between align-items-center card-text"
            data-testid="topic-points-title"
          >
            Prayer Points
            <svg
              class="text-secondary fw-bold"
              fill="currentColor"
              height="30px"
              stroke="currentColor"
              stroke-width="0"
              style="cursor: pointer;"
              viewBox="0 0 16 16"
              width="30px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z"
              />
              <path
                d="M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z"
              />
            </svg>
          </p>
          <p
            class="d-flex align-items-center gap-2 text-secondary card-text"
          >
            <svg
              fill="currentColor"
              height="20px"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 640 512"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M272 191.91c-17.6 0-32 14.4-32 32v80c0 8.84-7.16 16-16 16s-16-7.16-16-16v-76.55c0-17.39 4.72-34.47 13.69-49.39l77.75-129.59c9.09-15.16 4.19-34.81-10.97-43.91-14.45-8.67-32.72-4.3-42.3 9.21-.2.23-.62.21-.79.48l-117.26 175.9C117.56 205.9 112 224.31 112 243.29v80.23l-90.12 30.04A31.974 31.974 0 0 0 0 383.91v96c0 10.82 8.52 32 32 32 2.69 0 5.41-.34 8.06-1.03l179.19-46.62C269.16 449.99 304 403.8 304 351.91v-128c0-17.6-14.4-32-32-32zm346.12 161.73L528 323.6v-80.23c0-18.98-5.56-37.39-16.12-53.23L394.62 14.25c-.18-.27-.59-.24-.79-.48-9.58-13.51-27.85-17.88-42.3-9.21-15.16 9.09-20.06 28.75-10.97 43.91l77.75 129.59c8.97 14.92 13.69 32 13.69 49.39V304c0 8.84-7.16 16-16 16s-16-7.16-16-16v-80c0-17.6-14.4-32-32-32s-32 14.4-32 32v128c0 51.89 34.84 98.08 84.75 112.34l179.19 46.62c2.66.69 5.38 1.03 8.06 1.03 23.48 0 32-21.18 32-32v-96c0-13.77-8.81-25.99-21.88-30.35z"
              />
            </svg>
            Sub
          </p>
          <ul
            class="fs-5"
            data-testid="topic-points-points"
          >
            <li
              class="my-3"
            >
              ab
            </li>
            <li
              class="my-3"
            >
              cd
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
